lang/src/org/partiql/lang/planner/transforms/optimizations/FilterScanToKeyLookup.kt

```diff
@@ -0,0 +1,308 @@
+package org.partiql.lang.planner.transforms.optimizations
+
+import com.amazon.ionelement.api.TextElement
+import com.amazon.ionelement.api.ionBool
+import com.amazon.ionelement.api.ionSymbol
+import org.partiql.lang.domains.PartiqlPhysical
+import org.partiql.lang.domains.toBindingCase
+import org.partiql.lang.errors.ProblemHandler
+import org.partiql.lang.eval.BindingName
+import org.partiql.lang.planner.PartiqlPhysicalPass
+import org.partiql.lang.planner.PlannerPipeline
+import org.partiql.lang.planner.StaticTypeResolver
+import org.partiql.lang.planner.transforms.DEFAULT_IMPL
+import org.partiql.lang.types.BagType
+import org.partiql.lang.types.ListType
+import org.partiql.lang.types.StructType
+
+/**
+ * The "filter scan to key lookup" pass identifies all equality expressions where either side is a primary key
+ * within the filter's predicate.  This class represents the identified key field and the opposing value expression of
+ * each such equality expression. For example, given the filter predicate: `foo.id = 42`, the [keyFieldName] is `id`
+ * and the [equivalentValue] is `(lit 42)`.
+ */
+data class FieldEqualityPredicate(val keyFieldName: String, val equivalentValue: PartiqlPhysical.Expr)
+
+/**
```

Comment:
I'll reword this somewhat, hopefully it will be clearer.  

However, I would really prefer to leave this kdoc here, even as long as it is, because it a) only applies to the Kotlin implementation of PartiQL, and b) will quickly rot because it'll be very easy to forget to update when this pass inevitably changes.